### PR TITLE
Add Window::is_maximized method

### PR DESCRIPTION
**Pull request by simast**
Related issues: #208 #1578

Currently, if you want to save window state to a persistent config (position, size, fullscreen/maximized flags etc) and then restore when user is relaunching your application - this is not possible for maximized window state at all. We have `Window::set_maximized` and `WindowBuilder::with_maximized` - but no way to query current state. Current maximized state can be changed outside of API (e.g. users clicking on maximize button in window frame).

Tested on Windows and macOS. This should probably be supported for Wayland/X11 - but I am not too familiar with the API and could not figure out how to implement it. Marked linux API as not implemented and set it to always return `false` for now. Maybe some linux platform maintainers could help get this functionality covered under this platform as well?

- [x] Tested on all platforms changed
(only Windows and macOS)
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [x] `cargo doc` builds successfully
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Review by msiglreith — COMMENTED** [review]
Thanks!
Looks good regarding the windows implementation (:

I'm waiting a bit before merged til other maintainers can have a look regarding API changes

**Comment by msiglreith** [comment]
@chrisduerr Could you check if the proposed API changes are fine?

**Comment by chrisduerr** [comment]
That's not really any of my business, you should probably contact @francesca64 for questions about cross-platform API design.

**Comment by simast** [comment]
Anything I can do to help get this merged?

**Comment by msiglreith** [comment]
Sorry, for the silence.
As there seems to be no complaints and the overall API change is quite minimal I would propose to go forward with this one (:
Could you fix the merge conflict and I would merge it in then - for the missing platform implementation I will open an issue

**Comment by simast** [comment]
👍 Just rebased, thanks.

**Review by msiglreith — APPROVED** [review]

**Pull request merged**
